float64 reward
float64 action